### ISSUE TITLE
Spurious error

### ISSUE THREAD

**Issue by cpitclaudel** [PyCQA/pylint#1098]
This program:

``` python
F = (lambda b: lambda: b + 7)(3)
print(F())
```

Produces this output:

```
$ python2.7 -m pylint -r n example.py 
No config file found, using default configuration
************* Module example
C:  1, 0: Missing module docstring (missing-docstring)
E:  1,23: Undefined variable 'b' (undefined-variable)

$ python3.5 -m pylint -r n example.py 
No config file found, using default configuration
************* Module example
C:  1, 0: Missing module docstring (missing-docstring)
E:  1,23: Undefined variable 'b' (undefined-variable)
```

Running it raises no errors, however:

```
$ python3.5 example.py 
10

$ python2.7 example.py 
10
```

Is the error spurious?

**Comment by patamushta** [comment]
After update pylint to 1.6.4 i have the same error.

**Comment by cpitclaudel** [comment]
I'm running 

```
pylint 1.6.1, 
astroid 1.4.7
Python 2.7.12 (default, Jul  1 2016, 15:12:24) 
[GCC 5.4.0 20160609]
```

Thanks for confirming :)

**Comment by degustaf** [comment]
Duplicate of issue #760.

**Comment by PCManticore** [comment]
Closed as duplicate of #760
